CoreWiki/Helpers/UrlHelpers.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System.Text;
+
+namespace CoreWiki.Helpers
+{
+    public class UrlHelpers
+    {
+
+        /// https://stackoverflow.com/questions/25259/how-does-stack-overflow-generate-its-seo-friendly-urls
+        /// <summary>
+        /// Produces optional, URL-friendly version of a title, "like-this-one". 
+        /// hand-tuned for speed, reflects performance refactoring contributed
+        /// by John Gietzen (user otac0n) 
+        /// </summary>
+        public static string URLFriendly(string title)
+        {
+            if (title == null) return "";
+
+            const int maxlen = 80;
+            int len = title.Length;
+            bool prevdash = false;
+            var sb = new StringBuilder(len);
+            char c;
+
+            for (int i = 0; i < len; i++)
+            {
+                c = title[i];
+                if ((c >= 'a' && c <= 'z') || (c >= '0' && c <= '9'))
+                {
+                    sb.Append(c);
+                    prevdash = false;
+                }
+                else if (c >= 'A' && c <= 'Z')
+                {
+                    // tricky way to convert to lowercase
+                    sb.Append((char)(c | 32));
+                    prevdash = false;
+                }
+                else if (c == ' ' || c == ',' || c == '.' || c == '/' ||
+                    c == '\\' || c == '-' || c == '_' || c == '=')
+                {
+                    if (!prevdash && sb.Length > 0)
+                    {
+                        sb.Append('-');
+                        prevdash = true;
+                    }
+                }
+                else if ((int)c >= 128)
+                {
+                    int prevlen = sb.Length;
+                    sb.Append(RemapInternationalCharToAscii(c));
+                    if (prevlen != sb.Length) prevdash = false;
+                }
+                if (i == maxlen) break;
+            }
+
+            if (prevdash)
+                return sb.ToString().Substring(0, sb.Length - 1);
+            else
+                return sb.ToString();
+        }
+
+
+        public static string RemapInternationalCharToAscii(char c)
+        {
```

Comment:
Do we really want to remove the international characters?

Reply:
The issue here was related to old browsers like Internet Explorer. As some people unfortunately still use Internet Explorer it might be worth keeping this.